File: sgi-webapp/src/app/core/models/csp/proyecto-anualidad-notificacion-sge.ts
import { I18nFieldValue } from '@core/i18n/i18n-field';
import { DateTime } from 'luxon';

export interface IProyectoAnualidadNotificacionSge {
  id: number;
  anio: number;
  proyectoFechaInicio: DateTime;
  proyectoFechaFin: DateTime;
  totalGastos: number;
  totalIngresos: number;
  proyectoId: number;
  proyectoTitulo: I18nFieldValue[];
  proyectoAcronimo: string;
  proyectoEstado: string;
  proyectoSgeRef: string;
  enviadoSge: boolean;
}

File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/dto/ProyectoAnualidadNotificacionSge.java
package org.crue.hercules.sgi.csp.dto;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.Collection;

import org.crue.hercules.sgi.csp.model.EstadoProyecto;
import org.crue.hercules.sgi.csp.model.ProyectoTitulo;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProyectoAnualidadNotificacionSge implements Serializable {
  private Long id;
  private Integer anio;
  private Instant proyectoFechaInicio;
  private Instant proyectoFechaFin;
  private BigDecimal totalGastos;
  private BigDecimal totalIngresos;
  private Long proyectoId;
  private Collection<ProyectoTitulo> proyectoTitulo;
  private String proyectoAcronimo;
  private EstadoProyecto proyectoEstado;
  private String proyectoSgeRef;
  private Boolean enviadoSge;
}
